docs/source/services/timelock_service/migration.rst

```diff
@@ -0,0 +1,110 @@
+
+.. contents::
+   :local:
+
+Step 1: Setting up the Timelock Server
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+First, setup the Timelock Server following the instructions in :ref:`timelock-installation`.
+
+Step 2: Determining the Atlas Timestamp
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+One can use the :ref:`clis` (specifically, the ``timestamp fetch`` command) to obtain a timestamp which is guaranteed
+to be greater than all timestamps issued so far. Make sure that you are requesting for a fresh timestamp as opposed to
+an immutable timestamp (the output should say ``The Fresh timestamp is: N``). Note down the value of this timestamp;
+from here on out, we'll refer to it as N.
+
+Step 3: Fast-Forwarding the Timelock Server
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Timelock Server exposes an administrative interface, which features a ``fast-forward`` endpoint. Note that this is
+not typically exposed to AtlasDB clients. One can use it to advance the timestamp on the Timelock Server to N, as
+follows:
+
+   .. code:: bash
+
+      curl -XPOST localhost:8080/test/timestamp-admin/fast-forward?newMinimum=N
+
+.. danger::
+
+   Make sure that N has been entered correctly.
+
+    - If N entered is too large, there will generally not be adverse consequences (apart from risks of ``long``
+      overflow), but...
+    - If N entered is too small and AtlasDB clients are restarted without rectifying this, this can result in
+      **SEVERE DATA CORRUPTION** because we lose the guarantee that timestamps are monotonically increasing.
+      As fast-forward is idempotent (it sets the timestamp to be the maximum of its current value and the
+      ``newMinimum``), *if clients have not been started again* this can be fixed by doing another fast-forward.
+
+Step 4: Invalidating the Old Atlas Timestamp
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The steps for invalidating the old Atlas timestamp will vary, depending on your choice of underlying key value store.
+
+- Dropping the table, generally speaking, will *not* work (on the next startup of an embedded Timestamp Service,
+  AtlasDB will believe it is starting up the Timestamp Service for the first time, and thus start again from 1).
+- Setting the value to ``Long.MAX_VALUE`` or ``Long.MIN_VALUE`` will not work (Java Longs do not throw on arithmetic
+  overflow, and although ordinarily the first timestamp AtlasDB issues is 1 we do not throw on negative numbers).
+- If using Postgres or Oracle, it suffices to rename the relevant column in the timestamp table (use ``ALTER TABLE``).
+- If using Cassandra, one method of invalidating the table is to overwrite the timestamp bound record with the
+  empty byte array (consider using ``cqlsh`` to do this):
+
+     .. code:: bash
+
+        INSERT INTO atlasdb."_timestamp" (key, column1, column2, value) VALUES (0x7473, 0x7473, -1, 0x);
+
+  You may want to save the old value of the timestamp, in the event that rollback is desired.
```

Comment:
I think we should just tell people that they should always save the old value, and give the command to get this.